### PR TITLE
fix(prometheus): disable auth for prometheus endpoint on mgmt listener

### LINKED ISSUE #9287: Authentication required for scraping prometheus metrics in emqx v4.4.10
### What happened?

According to #8955, I thought authentication was no longer required, but it was.

Am I mistaken?

### What did you expect to happen?

Authentication for `api/v4/emqx_prometheus` was not required.

### How can we reproduce it (as minimally and precisely as possible)?

<details>

call `api/v4/emqx_prometheus` without authentication

```console
$ curl -i http://127.0.0.1:8081/api/v4/emqx_prometheus?type=prometheus
HTTP/1.1 401 Unauthorized
WWW-Authenticate: Basic Realm="minirest-server".
content-length: 12
Date: Wed, 02 Nov 2022 12:34:49 GMT
server. Cowboy

UNAUTHORIZED.
```

call `api/v4/emqx_prometheus` with authentication

```console
$ curl -i http://admin:public@127.0.0.1:8081/api/v4/emqx_prometheus?type=prometheus
http/1.1 200 ok
Content Length: 10685
Content Type: text/plain
Date: Wed, 02 Nov 2022 12:34:54 GMT
Connect to the server. Cowboy.

# TYPE erlang_vm_ets_limit gauge
erlang_vm_ets_limit 2097152
# TYPE erlang_vm_logical_processors gauge
erlang_vm_logical_processors 4
...
```

</details>

### Anything else we need to know?

_No response_

### EMQX version

<details>

```console
/opt/emqx $ emqx_ctl broker
sysdescr  : EMQ X Broker
version   : 4.4.10
uptime    : 42 minutes, 19 seconds
datetime  : 2022-11-02 12:43:13
```

</details>

### OS version

Docker image: emqx:4.4.10

### Log files

_No response_

### PR DESCRIPTION
**Pull request by JimMoen**
Fixes #9287 

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/` dir
- [ ] For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)
- [ ] For internal contributor: there is a jira ticket to track this change, and another jira tickt to track doc updates (if any)
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

### CODE REVIEW

File: apps/emqx_management/src/emqx_mgmt_http.erl [hunk 1]
```diff
@@ -124,6 +124,11 @@ handle_request(_Method, _Path, Req) ->
     cowboy_req:reply(400, #{<<"content-type">> => <<"text/plain">>}, <<"Not found.">>, Req).
 
 authorize_appid(Req) ->
+    authorize_appid(binary_to_list(cowboy_req:path(Req)), Req).
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_http.erl:127]:
```suggestion
    authorize_appid(cowboy_req:path(Req), Req).
```

File: apps/emqx_management/src/emqx_mgmt_http.erl [hunk 2]
```diff
@@ -124,6 +124,11 @@ handle_request(_Method, _Path, Req) ->
...
+
+authorize_appid("/api/v4/emqx_prometheus", _Req) ->
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_http.erl:129]:
```suggestion
authorize_appid(<<"/api/v4/emqx_prometheus">>, _Req) ->
```

**Comment by zmstone** [comment]
The original PR merged too soon, 
we need to make it right this time:  Please make sure the method is 'GET'

**Comment by JimMoen** [comment]
> The original PR merged too soon, we need to make it right this time: Please make sure the method is 'GET'

Methods other than `GET` will return 404 not found. See [Prometheus API](https://github.com/emqx/emqx/blob/v4.3.21/apps/emqx_prometheus/src/emqx_prometheus.erl#L30-#L35)

image

**Review by zmstone — APPROVED** [review]

**Pull request merged**